### DISCUSSION TITLE
How long do CNAME records take to propagate?

### DISCUSSION THREAD

**Discussion by bigsee** [vercel/vercel#4069] [General] [answered]
Hello!

I am trying to add CNAME records to validate my domain with Sendgrid.

Trying to validate within Sendgrid returns the following error: `Expected CNAME record for "xxxxx.mydomain.com" to match "sendgrid.net", but got "alias.zeit.co.".` Running `now dns ls my.domain.com` shows the all CNAME records including "xxxxx" with a value of "sendgrid.net.".
(The first CNAME record is a wildcard with the value of "alias.zeit.co.".)

I realise there is an extra "." but my understanding is that is required when adding the DNS records. Checking propagation at https://www.whatsmydns.net/ reveals that the DNS has not propagated.

Any idea who long I should expect to wait for CNAME records to propagate?

**Comment by coetry** [comment, marked as answer]
@bigsee DNS propagation is always a tricky subject since there are numerous factors that can cause a delay. It's a massive distributed system with many parties involved! 

We are however, working on a tool to better help visualize the DNS journey from our dashboard, so stay tuned! 

If you continue having issues though, send us an email at support@zeit.co and we can take a closer look at your specific situation.

**Reply by bigsee** [reply]
Thank you for the info! In the meantime the propagation completed.

I was just confused for a couple of hours as the Now CLI told me the records were created and I didn't know whether I'd done something wrong or should expect to wait a while. Other times I've created new CNAME records and they have been detected almost instantly. I guess it is just luck of the draw.

Look forward to seeing the new tool!
